File: Data Structures & Algorithms/two-integer-sum/submission-10.cpp
#include <algorithm> // std::sort
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        using namespace std;
        vector<pair<int, int>> pairs;
        for (int i = 0; i < nums.size(); i ++) {
            pairs.push_back({nums[i], i});
        }
        sort(pairs.begin(), pairs.end(), [](const auto& a, const auto& b) {
            return a.first < b.first;
        });
        int i = 0;
        int j = pairs.size() - 1;
        while (i < j) {
            const int s = pairs[i].first + pairs[j].first;
            if (s > target) {
                j -= 1;
            } else if (s < target) {
                i += 1;
            } else {
                break;
            }
        }
        vector<int> m = {pairs[i].second, pairs[j].second};
        sort(m.begin(), m.end());
        return m;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-10.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums.sort()
        prev = nums[0]
        ans = 0
        for n in nums[1:]:
            inc = n - prev
            if inc > 1:
                break
            else:
                ans += 1
                prev += inc
        return ans